src/components/CreateShareStructure/ShareStructure.vue

```diff
@@ -217,9 +217,10 @@ export default class ShareStructure extends Mixins(CurrencyLookupMixin) {
     let rules: Array<Function> = []
     if (!this.hasNoMaximumShares) {
       rules = [
-        (v: string) => (v !== null && v !== '') || 'Number of shares is required',
+        (v: string) => !!v || 'Number of shares is required',
         (v: string) => /^-?\d+$/.test(v) || 'Must be a whole number',
-        (v: string) => +v > 0 || 'Number must be greater than 0'
+        (v: string) => (+v > 0) || 'Number must be greater than 0',
+        (v: string) => (v && v.toString().length < 16) || 'Number must be less than 16 digits'
```

Comment:
I had to cast v to a string as we sometimes set the value to a number in the unit tests.